clients/src/test/java/org/apache/kafka/clients/producer/MockProducerTest.java

```diff
@@ -47,22 +47,21 @@
 public class MockProducerTest {
 
     private final String topic = "topic";
-    private final MockProducer<byte[], byte[]> producer = new MockProducer<>(true, new MockSerializer(), new MockSerializer());
     private final ProducerRecord<byte[], byte[]> record1 = new ProducerRecord<>(topic, "key1".getBytes(), "value1".getBytes());
     private final ProducerRecord<byte[], byte[]> record2 = new ProducerRecord<>(topic, "key2".getBytes(), "value2".getBytes());
-
-
+    
     @Test
-    @SuppressWarnings("unchecked")
     public void testAutoCompleteMock() throws Exception {
-        Future<RecordMetadata> metadata = producer.send(record1);
-        assertTrue("Send should be immediately complete", metadata.isDone());
-        assertFalse("Send should be successful", isError(metadata));
-        assertEquals("Offset should be 0", 0L, metadata.get().offset());
-        assertEquals(topic, metadata.get().topic());
-        assertEquals("We should have the record in our history", singletonList(record1), producer.history());
-        producer.clear();
-        assertEquals("Clear should erase our history", 0, producer.history().size());
+        try (MockProducer<byte[], byte[]> producer = new MockProducer<>(true, new MockSerializer(), new MockSerializer())) {
```

Comment:
Have to say it's a bit annoying to have all these diffs just to close the mock producer (which does not have any resources to close). I wonder if we can find a lighter pattern. What I was trying to suggest before was to continue to use the shared `producer` with an `@After`. We can use a factory method to create it in each method. For example:
```java
@After
public void cleanup() {
  if (this.producer != null) this.producer.close()
}

private void buildMockProducer(boolean autoComplete) {
  this.producer = new MockProducer<>(autocomplete, new MockSerializer(), new MockSerializer()))
}
```
We can add overloads to `buildMockProducer` as needed. What do you think?